### DISCUSSION TITLE
Nginx compilation issue with the waflz module

### DISCUSSION THREAD

**Discussion by DejanPelzel** [Edgio/waflz#212] [Q&A] [answered]
Hello 👋,

Stumbled upon Waflz yesterday and it seems really promising. I thought I would do some testing, however, I've been struggling to get the nginx module to build on Ubuntu 18 (and 16).

I thought I would ask if there's anyone that successfully did this or experienced a similar issue and perhaps could share a pointer?

To simplify this, I'm trying quite a simple build
```
./configure --prefix=/etc/nginx \
--sbin-path=/usr/sbin/nginx \
--conf-path=/etc/nginx/nginx.conf \
--error-log-path=/var/log/nginx/error.log \
--http-log-path=/var/log/nginx/access.log \
--pid-path=/var/run/nginx.pid \
--lock-path=/var/run/nginx.lock \
--user=nginx \
--group=nginx \
--with-pcre \
--with-pcre-jit \
--with-threads \
--add-module=../waflz/plugins/nginx

make -j16
```

Unfortunately, however, the build fails:

```
In file included from ../waflz/plugins/nginx/../../include/waflz/def.h:30:0,
                 from ../waflz/plugins/nginx/../../include/waflz/rqst_ctx.h:15,
                 from ../waflz/plugins/nginx/../../include/waflz/scopes.h:30,
                 from ../waflz/plugins/nginx/src/ngx_http_waflz_module.h:36,
                 from ../waflz/plugins/nginx/src/ngx_http_waflz_module.c:30:
../waflz/plugins/nginx/../../include/waflz/city.h:67:10: fatal error: utility: No such file or directory
```

The actual waflz or nginx compile well by themselves otherwise.

If utility is added to the path, it proceeds to the compilation of waflz, but that then fails with "error: unknown type name ‘namespace’".

It seems the issue steems down to compiling with C while waflz uses C++. I tried a couple of things, compiling with c++ instead, but just end up with other errors.

**Comment by dev0z** [comment, marked as answer]
Check out [nginx_deps](https://github.com/VerizonDigital/waflz/commit/6bb4521dc4bfdc4ce1ddd96a9c141fd36ee4a0e8) branch. It has the fix for the error you are getting. The Nginx plugin has kind of fallen behind the changes in master. But the good news is we are going to pick up the development soon.
